File: fix_form.js
const fs = require('fs');
const path = 'C:\\Users\\jhose\\OneDrive\\Desktop\\PROYECTO DE CODIGO\\LEGADO_BIBLICO_PROD\\data_iglesia_v1.js';

let content = fs.readFileSync(path, 'utf-8');

// The old code to replace (from the button write to the closing };)
const oldCode = content.substring(
    content.indexOf("    ventana.document.write('<button class=\"btn\" id=\"btnEnviar\">"),
    content.indexOf("};", content.indexOf("ventana.document.getElementById('btnEnviar')")) + 2
);

const newCode = `    ventana.document.write('<button class="btn" id="btnEnviar">\u{1f4e4} ENVIAR PROGRAMA</button></div>');
    ventana.document.write('<div class="ft">Generado por Legado B\u00edblico</div></div></body></html>');
    ventana.document.close();

    // Inyectar script con createElement (funciona cross-window)
    var sc = ventana.document.createElement('script');
    sc.textContent = 'function enviarForm(){' +
        'var g=function(i){var e=document.getElementById("f-"+i);return e&&e.value.trim()?e.value.trim():"---"};' +
        'var L="\\n";' +
        'var t="\\u26ea *PROGRAMA DE CULTO*"+L' +
        '+"\\ud83d\\udcc5 *Fecha:* "+g("fecha")+L' +
        '+"\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501"+L' +
        '+"\\u2728 Doxolog\\u00eda: "+g("doxologia")+L' +
        '+"\\ud83d\\ude4f Invocaci\\u00f3n: "+g("invocacion")+L' +
        '+"\\ud83d\\ude4c Bienvenida: "+g("bienvenida")+L' +
        '+"\\ud83d\\udc76 Rinc\\u00f3n Infantil: "+g("infantil")+L' +
        '+"\\ud83d\\udcb0 *Diezmos y Ofrendas:*"+L' +
        '+"   \\ud83e\\uddd4 Di\\u00e1cono: "+g("diacono")+L' +
        '+"   \\ud83d\\udc69 Diaconisa: "+g("diaconisa")+L' +
        '+"\\ud83c\\udfb5 Himno de Adoraci\\u00f3n: "+g("himno")+L' +
        '+"\\ud83d\\udcd6 Lectura B\\u00edblica: "+g("lectura")+L' +
        '+"\\ud83d\\ude4f Oraci\\u00f3n Intercesora: "+g("oracion")+L' +
        '+"\\ud83c\\udfa4 M\\u00fasica Especial: "+g("musica")+L' +
        '+"\\ud83c\\udf99\\ufe0f Presentar al Predicador(a): "+g("presentar")+L' +
        '+"\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501"+L' +
        '+"\\u26ea *Predicador:* "+g("predicador")+L' +
        '+"\\ud83d\\udcdd *Tema:* "+g("tema")+L' +
        '+"\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501\\u2501"+L' +
        '+"\\ud83c\\udfb6 Himno Final: "+g("himnoFinal")+L' +
        '+"\\ud83e\\udd32 Oraci\\u00f3n Final: "+g("oracionFinal")+L' +
        '+L+"\\ud83d\\udcf1 _Legado B\\u00edblico_";' +
        'if(navigator.share){navigator.share({title:"Programa de Culto",text:t}).catch(function(){})}' +
        'else{navigator.clipboard.writeText(t).then(function(){alert("\\ud83d\\udccb Copiado! Pegalo donde quieras.")}).catch(function(){prompt("Copia este texto:",t)})}' +
        '}';
    ventana.document.body.appendChild(sc);
    ventana.document.getElementById('btnEnviar').addEventListener('click', function(){ ventana.enviarForm(); });
};`;

if (oldCode.length > 50) {
    content = content.replace(oldCode, newCode);
    fs.writeFileSync(path, content, 'utf-8');
    console.log('OK - replaced ' + oldCode.length + ' chars with ' + newCode.length + ' chars');
} else {
    console.log('ERROR - old code not found (length: ' + oldCode.length + ')');
}

File: fix_form.py
import re

path = r"C:\Users\jhose\OneDrive\Desktop\PROYECTO DE CODIGO\LEGADO_BIBLICO_PROD\data_iglesia_v1.js"

with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

# Find the section to replace: from the btn-enviar line to the closing };
old_pattern = r"(    ventana\.document\.write\('<button class=\"btn\".*?ENVIAR PROGRAMA.*?\n)(.*?)(^\};$)"
# This won't work well with regex. Let's do it by finding markers.

lines = content.split('\n')
new_lines = []
skip = False
found = False

for i, line in enumerate(lines):
    if skip:
        # Look for the closing }; of enviarFormularioVacioWA
        if line.strip() == '};':
            skip = False
            found = True
            # Write replacement
            new_lines.append("    ventana.document.write('<button class=\"btn\" id=\"btnEnviar\">\U0001f4e4 ENVIAR PROGRAMA</button></div>');")
            new_lines.append("    ventana.document.write('<div class=\"ft\">Generado por Legado B\u00edblico</div></div></body></html>');")
            new_lines.append("    ventana.document.close();")
            new_lines.append("")
            new_lines.append("    // Inyectar funci\u00f3n directamente (sin script embebido)")
            new_lines.append("    ventana.enviarForm = function() {")
            new_lines.append("        var doc = ventana.document;")
            new_lines.append("        var g = function(id) { var e = doc.getElementById('f-' + id); return e && e.value.trim() ? e.value.trim() : '---'; };")
            new_lines.append("        var lineas = [")
            new_lines.append("            '\u26ea *PROGRAMA DE CULTO*',")
            new_lines.append("            '\U0001f4c5 *Fecha:* ' + g('fecha'),")
            new_lines.append("            '\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501',")
            new_lines.append("            '\u2728 Doxolog\u00eda: ' + g('doxologia'),")
            new_lines.append("            '\U0001f64f Invocaci\u00f3n: ' + g('invocacion'),")
            new_lines.append("            '\U0001f64c Bienvenida: ' + g('bienvenida'),")
            new_lines.append("            '\U0001f476 Rinc\u00f3n Infantil: ' + g('infantil'),")
            new_lines.append("            '\U0001f4b0 *Diezmos y Ofrendas:*',")
            new_lines.append("            '   \U0001f9d4 Di\u00e1cono: ' + g('diacono'),")
            new_lines.append("            '   \U0001f469 Diaconisa: ' + g('diaconisa'),")
            new_lines.append("            '\U0001f3b5 Himno de Adoraci\u00f3n: ' + g('himno'),")
            new_lines.append("            '\U0001f4d6 Lectura B\u00edblica: ' + g('lectura'),")
            new_lines.append("            '\U0001f64f Oraci\u00f3n Intercesora: ' + g('oracion'),")
            new_lines.append("            '\U0001f3a4 M\u00fasica Especial: ' + g('musica'),")
            new_lines.append("            '\U0001f399\ufe0f Presentar al Predicador(a): ' + g('presentar'),")
            new_lines.append("            '\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501',")
            new_lines.append("            '\u26ea *Predicador:* ' + g('predicador'),")
            new_lines.append("            '\U0001f4dd *Tema:* ' + g('tema'),")
            new_lines.append("            '\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501',")
            new_lines.append("            '\U0001f3b6 Himno Final: ' + g('himnoFinal'),")
            new_lines.append("            '\U0001f932 Oraci\u00f3n Final: ' + g('oracionFinal'),")
            new_lines.append("            '',")
            new_lines.append("            '\U0001f4f1 _Legado B\u00edblico_'")
            new_lines.append("        ].join('\\n');")
            new_lines.append("        if (ventana.navigator.share) {")
            new_lines.append("            ventana.navigator.share({title: 'Programa de Culto', text: lineas}).catch(function(){});")
            new_lines.append("        } else {")
            new_lines.append("            ventana.navigator.clipboard.writeText(lineas).then(function(){")
            new_lines.append("                ventana.alert('\U0001f4cb Copiado! P\u00e9galo donde quieras enviar.');")
            new_lines.append("            }).catch(function(){")
            new_lines.append("                ventana.prompt('Copia este texto:', lineas);")
            new_lines.append("            });")
            new_lines.append("        }")
            new_lines.append("    };")
            new_lines.append("    ventana.document.getElementById('btnEnviar').onclick = ventana.enviarForm;")
            new_lines.append("};")
        continue
    
    # Detect the button line (start of section to replace)  
    if 'ENVIAR PROGRAMA' in line and 'ventana.document.write' in line and 'btn' in line:
        skip = True
        continue
    
    new_lines.append(line)

if found:
    with open(path, 'w', encoding='utf-8') as f:
        f.write('\n'.join(new_lines))
    print(f"OK - {len(new_lines)} lines written")
else:
    print("ERROR - pattern not found")
